netty/src/test/java/io/grpc/netty/NettyClientStreamTest.java

```diff
@@ -436,7 +433,10 @@ public void removeUserAgentFromApplicationHeaders() {
     metadata.put(GrpcUtil.USER_AGENT_KEY, "bad agent");
     listener = mock(ClientStreamListener.class);
     Mockito.reset(writeQueue);
-    when(writeQueue.enqueue(any(QueuedCommand.class), any(boolean.class))).thenReturn(future);
+    ChannelPromise completedPromise = new DefaultChannelPromise(channel)
+        .setSuccess();
+    when(writeQueue.enqueue(any(QueuedCommand.class), any(boolean.class)))
+        .thenReturn(completedPromise);
```

Comment:
I want to return a future that will complete. The mocking is horrible, and I don't want tests to just hang in the future because the code changes slightly.